src/owp_snow_obs/owp_snow_obs.py

```diff
@@ -0,0 +1,255 @@
+        # Unstack is not reproducible, must sort after
+        obs_df = (
+            obs_df
+            .set_index(index_cols2)
+            .sort_index()
+            .reset_index())
+
+        self.attr_data['ref_date_time'] = (
+            pd.to_datetime(obs_df.datetime[0])
+            .round('D')
+            .strftime('%Y-%m-%dT%H:%M:%SZ'))
+
+        if self.thin_swe == 1.0 and self.thin_depth == 1.0:
+            print('No output requested, but variables thined by 100%')
+            return(None)
+        self.attr_data['thin_swe'] = self.thin_swe
+        self.attr_data['thin_depth'] = self.thin_depth
+        thin_dict = {
+            'snow_water_equivalent_mm': self.thin_swe,
+            'snow_depth_m': self.thin_depth}
+        # to make it reproducible if a seed is not passed
+        if self.thin_random_seed is None:
+            # Set a seed for each day - reproducibly random
+            time_datum = pd.Timestamp('1979-01-01 00:00:00', tz='UTC')
+            time_diff = pd.Timestamp(self.attr_data['ref_date_time']) - time_datum
+            self.thin_random_seed = int(time_diff.total_seconds())
+        np.random.seed(self.thin_random_seed)
+        for var, thin in thin_dict.items():
+            if thin > 0.0:
+                wh_var = np.where(~np.isnan(obs_df[f'ObsValue {var}']))[0]  # 1-D
+                randoms = np.random.uniform(size=len(wh_var))
+                thin_quantile = np.quantile(randoms, thin)
+                thin_inds = np.where(randoms <= thin_quantile)[0]
+                for cc in data_cols:
+                    obs_df.loc[wh_var[thin_inds], f'{cc} {var}'] = np.nan
+                if all(np.isnan(obs_df[f'ObsValue {var}'])):
+                    # If there's no data, drop the variable(s) from the output list
+                    del output_var_dict[f'{var}']
+                    # Also can then remove nans in the remaining other var (since we only have 2 vars)
+                    var_other = list(output_var_dict.keys())[0]
+                    wh_not_var_other = np.where(np.isnan(obs_df[f'ObsValue {var_other}']))[0]  # 1-D
+                    obs_df = obs_df.drop(wh_not_var_other).reset_index()
+                    # does this work when both variables are thinned by some amount?
+
+        self.data[('datetime', 'MetaData')] = obs_df.datetime.values
+        self.data[('latitude', 'MetaData')] = obs_df.latitude.values
+        self.data[('longitude', 'MetaData')] = obs_df.longitude.values
+
+        for obs_var, ioda_var in output_var_dict.items():
+            # define the ioda variable
+            self.var_dict[ioda_var]['valKey'] = ioda_var, iconv.OvalName()
+            self.var_dict[ioda_var]['errKey'] = ioda_var, iconv.OerrName()
+            self.var_dict[ioda_var]['qcKey'] = ioda_var, iconv.OqcName()
+            # define ioda meta/ancillary
+            self.units[ioda_var] = output_var_unit_dict[ioda_var]
+            self.var_metadata[ioda_var]['coordinates'] = 'longitude latitude'
```

Comment:
Please add an attribute name "units" to the ObsValue, ObsError and PreQC versions of these variables. It looks like you have the units already in the self.units dictionary so you can use those values for the ObsValue and ObsError versions of the variables. For categorical data (such as PreQC) use "unitless" for the units. This will signal to downstream UFO operators that these variables cannot be used in math formulas. Thanks!

Reply:
We need to add station elevation as "height" in m to output, and I should check variable names for existing conventions, especially for snow depth

Reply:
Also added units for new metadata fields.